CRM/Contribute/Form/Contribution.php

```diff
@@ -369,7 +369,7 @@ public function setDefaultValues() {
       $defaults['total_amount'] = CRM_Utils_Money::format($total_value, NULL, '%a');
       if (!empty($defaults['tax_amount'])) {
         $componentDetails = CRM_Contribute_BAO_Contribution::getComponentDetails($this->_id);
-        if (!(CRM_Utils_Array::value('membership', $componentDetails) || CRM_Utils_Array::value('participant', $componentDetails))) {
+        if (!(!empty($componentDetails['membership']) || !empty($componentDetails['participant']))) {
```

Comment:
This might be clearer as `empty($componentDetails['membership']) && empty($componentDetails['participant'])`